test/kv.go

```diff
@@ -1124,6 +1120,43 @@ func cas(kv kvdb.Kvdb, t *testing.T) {
 	assert.NoError(t, err, "CompareAndSet should succeed on an correct value and modified index")
 }
 
+func cad(kv kvdb.Kvdb, t *testing.T) {
+	fmt.Println("\ncad")
+
+	key := "foo/docker"
+	val := "great"
+	defer func() {
+		kv.DeleteTree(key)
+	}()
+
+	kvPair, err := kv.Put(key, []byte(val), 0)
+	assert.NoError(t, err, "Unxpected error in Put")
+
+	kvPair, err = kv.Get(key)
+	assert.NoError(t, err, "Failed in Get")
+
+	copyKVPair1 := *kvPair
+	copyKVPair1.Value = []byte("badval")
+	_, err = kv.CompareAndDelete(&copyKVPair1, kvdb.KVFlags(0))
+	assert.Error(t, err, "CompareAndDelete should fail on an incorrect previous value")
+
+	copyKVPair2 := *kvPair
+	copyKVPair2.ModifiedIndex++
+	_, err = kv.CompareAndDelete(&copyKVPair2, kvdb.KVModifiedIndex)
+	assert.Error(t, err, "CompareAndDelete should fail on an incorrect modified index")
+
+	//kvPair.ModifiedIndex--
```

Comment:
nit: remove commented code, just kvPair to delete instead of copy ?

Reply:
as we are modifying the indexes and values, hence we are taking a copy.